lib/customer-deposit-wallet/test/scenario/Test/Scenario/Wallet/Deposit/Exchanges.lhs.md

```diff
@@ -0,0 +1,218 @@
+
+The function `getCustomerHistory` returns a `TxSummary` for each transaction that is related to this customer. For every `TxSummary`, the `received` field records the total deposit made by the customer at this address in this transaction.
+
+(The `spent` field has informative purpose only, and records whether the wallet has moved any funds out of this address.)
+
+The following scenario illustrates how `getCustomerHistory` records deposits:
+
+```haskell
+scenarioTrackDepositOne
+    :: ScenarioEnv -> WalletInstance -> IO ()
+scenarioTrackDepositOne env w = do
+    address <- Wallet.createAddress customer w
+
+    -- no deposits
+    txsummaries0<- Wallet.getCustomerHistory customer w
+    assert $ txsummaries0 == []
+
+    -- first deposit
+    depositFundsAt env address coin
+    txsummaries1 <- Wallet.getCustomerHistory customer w
+    assert $ map (received . transfer) txsummaries1 == [coin]
+
+    -- second deposit
+    depositFundsAt env address coin
+    txsummaries2 <- Wallet.getCustomerHistory customer w
+    assert $ map (received . transfer) txsummaries2 == [coin, coin]
+  where
+    customer = 7 :: Customer
+    coin = ada 12
+```
+
+## 3. Track deposits at all addresses
+
+A centralized exchange typically wants to monitor all transactions in a recent time window for activity in order to synchronize customer deposits on the blockchain ledger with the exchange ledger recording customer balances.
+
+This is a task for the `getCustomerHistories` function — it returns a mapping from customers to `TxSummaries` that record the entire activity within the given time interval.
+
+The time interval is specified by a `from` and a `to` point on the blockchain. We note that the `from` argument is exclusive while the `to` argument is inclusive.
+We use the type `ChainPoint` to specify points on the blockchain — this type uses both a slot number and a block header to uniquely identify a block. We do this in order to allow atomic operations — in the event that the `to` or `from` point are no longer part of the consensus chain, the `getCustomerHistories` functions throws an exception.
+
+The wallet is synchronized to a particular point on the blockchain — use `getWalletTip` to query it.
+
+```haskell
+scenarioTrackDepositAll
+    :: ScenarioEnv -> WalletInstance -> IO ()
+scenarioTrackDepositAll env w = do
+    address1 <- Wallet.createAddress customer1 w
+    address2 <- Wallet.createAddress customer2 w
+
+    from <- Wallet.getWalletTip w
+    depositFundsAt env address1 coin
+    depositFundsAt env address2 coin
+    depositFundsAt env address1 (coin <> coin)
+    to <- Wallet.getWalletTip w
+
+    history <- Wallet.getCustomerHistories (from,to) w
+    assert $
+        Map.map received history
+      ==
+        Map.fromList
+            [ (customer1, coin <> coin <> coin)
+            , (customer2, coin)
+            ]
+  where
+    customer1, customer2 :: Customer
+    customer1 = 1
+    customer2 = 2
+    coin = ada 3
+```
```

Comment:
what about adding the following property at this point? Below `(to1=from2)`
``` 
history1 <-  getCustomerHistories(from1,to1) w
history2 <- getCustomerHistories(from2,to2) w
history3 <- getCustomerHistories(from1,to2) w
history1 union history2 == history3
```
With adequately defined `union`

Reply:
> what about adding the following property at this point? Below (to1=from2)

I agree that this would be a good unit test and/or needs to be specified in some form.

However, the focus of these scenario tests is documentation / tutorial-style exposition, and I believe that the range argument is documented well enough in text form, so that a reader can infer this property from the text.